### PR TITLE
Add phabox

### PR DESCRIPTION
**Pull request by bernt-matthias**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/phabox/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  version: "{{ version }}"
+
+source:
+  - url: https://github.com/KennthShang/{{ name }}/archive/refs/tags/v{{ version }}-beta.tar.gz
+    sha256: 206d7c5fef0b9cc5fa51e1ed66c77c1f53da47bbdaed1e6f2c0058f8a8aaa403
+    patches:
+      - imports.patch
+  # add setup.py based on https://github.com/KennthShang/PhaBOX/blob/243f78aa82f682c9b8370027c882a4658b9d2945/setup.py
+  # added package_data and entry_points
+  - path: setup.py
```
Comment (bgruening) [inline recipes/phabox/meta.yaml]: Do this in a build.sh and put comments in there. You can copy the setup.py and then run `"{{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vvv"`

File: recipes/phabox/setup.py
Comment (bgruening) [inline recipes/phabox/setup.py:1]: Is that still needed?

File: recipes/phabox/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  run_exports:
+    - {{ pin_subpackage('phabox', max_pin="x.x") }}
+
+requirements:
+  host:
+    - pip
+    - python >=3
+  run:
+    - python >=3
+    - biopython >= 1.81
```
Comment (bernt-matthias) [inline recipes/phabox/meta.yaml:27]:
```suggestion
    - biopython >=1.81
```

**Comment by bernt-matthias** [comment]
That's odd @bgruening. 

After you pushed [879e019](https://github.com/bioconda/bioconda-recipes/pull/50753/commits/879e01951de379b7ea38a32b2a32fb3071086d01) tests ran (and revealed a problem) .. but afterwards they are cancelled again...

**Comment by bgruening** [comment]
There is something else going on. I haven't seen this before. The only idea I have it making the dependencies more strict to help the solver or taking one dependencies after the other out and see which one it is.

**Comment by bgruening** [comment]
@bernt-matthias upstream released version 2.0. Do you want to update to version 2 before we try here more?

File: recipes/phabox/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{% set name = "PhaBOX" %}
+{% set version = "1.0.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/KennthShang/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 2e8ff502af4633edad2635ba8c7c595d2f6c13482c1b7fb98bfd01218fedd2f8
```
Comment (bernt-matthias) [inline recipes/phabox/meta.yaml:10]:
```suggestion
  sha256: 3e120da4b6c37d1ae9f7291382365012a4efa0c4e704068c1ecd897e5fd8178a
```

File: recipes/phabox/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{% set name = "PhaBOX" %}
+{% set version = "1.0.0" %}
```
Comment (bernt-matthias) [inline recipes/phabox/meta.yaml:2]: {% set version = "2.0.0" %}

**Comment by bernt-matthias** [comment]
Magic :)

Slightly unsatisfying, since  I would have liked to know what was going on :)

**Comment by bgruening** [comment]
The patch is applying as it is ...? mh ...

**Review by bgruening — APPROVED** [review]

**Pull request merged**